static/js/services/xml-forms.js

```diff
@@ -60,6 +60,10 @@ angular.module('inboxServices').factory('XmlForms',
     };
 
     var filter = function(form, options, user) {
+      if (!options.includeCollect && form.context && form.context.collect) {
```

Comment:
This looks fine, unless we ever have a form which is both for collect and for in-app use.  I doubt we currently have any shared forms, and it seems unlikely in the future too.